chore: Refactor structure of Cookbook

novusnota opened the pull request:
Shiny:

![image](https://github.com/tact-lang/tact-docs/assets/68142933/bd2a0b58-e4e0-4608-aac0-95f3ec88ccad)

* Splitted the main page into many categories
* Added dedicated pages for examples with DEXes: #146, #149
* Deleted basics (`if...else` statements, loops, etc.)
* Added page: Algorithms (for code samples from past contests and for future contests, practical things or general proofs-of-concept for competitive programming and beyond)

Closes #142 (refactor)
Closes #141 (removed basics)
Closes #161 (added an algorithms page, moved examples with maps onto a special data structures page)

anton-trunov commented on pages/cookbook/index.mdx:
```diff
@@ -1,932 +1,96 @@
+<Cards>
+  <Cards.Card
+    icon={<DiagramIcon />}
+    title="Mass, multi-contract communication"
+    href="/cookbook/mass-communication"
```
```suggestion
    title="Multi-contract communication"
    href="/cookbook/multi-communication"
```

anton-trunov commented on pages/cookbook/primitive-types.mdx:
```diff
@@ -0,0 +1,243 @@
+# Primitive types
```
the primitive types section is not really a collection of recipes, so its contents needs to be eventually moved to the Book sections

novusnota replied:
Agreed

anton-trunov commented on pages/cookbook/primitive-types.mdx:
```diff
@@ -0,0 +1,243 @@
+
+## `Int` ↔ `String` [#int-and-string]
+
+### How to convert a `String` to an `Int`
```
this one we don't need to move to Book, because it looks kind of like a recipe (although to make it self-contained, we need to create a function of type `String -> Int?`  with this functionality)

novusnota replied:
Will do. Also: https://github.com/tact-lang/tact-docs/issues/168#issuecomment-2037001102

anton-trunov commented on pages/cookbook/misc.mdx:
```diff
@@ -0,0 +1,59 @@
+
+</Callout>
+
+## How to generate a random number
```
this section looks like a duplicate of `cookbook/random`

anton-trunov reviewed (changes requested):
good stuff!

novusnota commented:
Temporarily archived examples with checking:
* Cells for equality (already covered in Operators), emptyness of Cells, whether they contain refs or bits
* Same stuff for Slices

This is done, such that later we could either:
* introduce a standalone page for Cells, Builders and Slices (with many links to TON Docs and brief layout of all the possibilities) 
* add such examples to Reference where they're suitable the most

Additionally, renamed "Primitive types" to "Type conversion", such that later we can populate it with more convoluted examples, if needed.

Plus, added an extension function for converting Strings to Ints. Note, that the description of [`.asSlice()`](https://docs.tact-lang.org/language/ref/strings#stringasslice) is very poor and the whole page needs a re-write :)
